### ISSUE TITLE
DetailsList Items not updating even if items are changed when usePageCache and useReducedRowRenderer props are set to true.

### ISSUE THREAD

**Issue by ahsanzahid-se** [microsoft/fluentui#14233]
### Environment Information

- **Package version(s)**: (6.112.0)
- **Browser and OS versions**: (Chrome latestest)

#### Actual behavior:
I am using the detailslist. I have introduced two new props ( usePageCache, useReducedRowRenderer ) to your list to increase my screen proformance. It increases proformance but when I changed my page from pagination. The items are changed but detailsList does not update the changed items.
#### Expected behavior:
Using usePageCache and useReducedRowRenderer props. When Items are changed the detailsList should also update the items.

### Priorities and help requested:

Are you willing to submit a PR to fix? (No)

Requested priority: (Blocking)

**Comment by paulgildea** [comment]
Thanks for the feedback @ahsanzahid-se !

If you could supply a codepen/codesandbox of the issue that you are running into that would help us investigate this issue. 

But to set expectations - we are only taking limited fixes to the `DetailsList` as it's a legacy component that is becoming more risky to fix. So this may end up becoming a won't fix if it requires updating the behavior of the `DetailsList` to enable the desired scenario to work.

Thanks!

**Comment by ahsanzahid-se** [comment]
@paulgildea 
Thanks for your feedback. Let me try again. I have a parent and a child component. Items to the detailslist is passed to the child from the parent. The child component also have items array which is available within its state. When the items are passed from the parent, I update the state (items) of the child component and break the reference also as mentioned by your official documentation. Now your detailslist get a new freshly created array of items in the props but still the detailslist does not show the new items I have passed.

**Comment by xugao** [comment]
@ahsanzahid-se - i tried setting both prop to true while changing the items work, here is the [codepen](https://codepen.io/xugao/pen/ZEQdPpz). pls provide a repro similar to this to demo the issue so we can help you
